src/expression.d

```diff
@@ -14159,14 +14159,19 @@ extern (C++) final class MinExp : BinExp
         {
             if (t2.ty == Tpointer)
             {
+                Type p1 = t1.nextOf();
+                Type p2 = t2.nextOf();
+
+                if (!p1.equivalent(p2))
+                {
+                    error("Operands must point to the same type (got %s and %s)", p1.toChars(), p2.toChars());
+                    return new ErrorExp();
+                }
+
                 // Need to divide the result by the stride
                 // Replace (ptr - ptr) with (ptr - ptr) / stride
                 d_int64 stride;
 
-                // make sure pointer types are compatible
-                if (Expression ex = typeCombine(this, sc))
```

Comment:
It just brings the types down to a common one, I couldn't find a case where something more than that was being done.